src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFiltering.cpp

```diff
@@ -93,9 +95,11 @@ namespace OpenMS
         mz_shifts_actual.push_back(std::numeric_limits<double>::quiet_NaN());
         mz_shifts_actual_indices.push_back(-1);
       }
-
     }
 
+    // early-out: not peaks found 
+    if (found_peaks < peaks_per_peptide_min_) return -1;
```

Comment:
Yes. I further speedup.

And the entire check can now be done within the method and not here
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringProfile.cpp#L199
Same here
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringCentroided.cpp#L164